src/client/resources/common.ts

```diff
@@ -50,3 +56,41 @@ export abstract class ClusterResource {
 function namedPath(listPath, name) {
   return [listPath, name].join('/');
 }
+
+export abstract class ClusterDiscoveryResource {
```

Comment:
RC: We don't want to call this `ClusterDiscoveryResource`. That could be confusing when seen in the context of kubernetes resources, which are usually prefixed as `Cluster` resources when they are cluster-scoped and non-namespaced.

I think simply `DiscoveryResource` as an abstract class works here.